src/execution_tree/primitives/row_set.cpp

```diff
@@ -0,0 +1,360 @@
+#include <numeric>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    ///////////////////////////////////////////////////////////////////////////
+    primitive create_row_set_operation(hpx::id_type const& locality,
+        std::vector<primitive_argument_type>&& operands,
+        std::string const& name, std::string const& codename)
+    {
+        static std::string type("set_row");
+        return create_primitive_component(
+            locality, type, std::move(operands), name, codename);
+    }
+
+    match_pattern_type const row_set_operation::match_data =
+    {
+        hpx::util::make_tuple("set_row",
+            std::vector<std::string>{"set_row(_1, _2, _3, _4, _5)"},
+            &create_row_set_operation,
+            &create_primitive<row_set_operation>)
+    };
+
+    ///////////////////////////////////////////////////////////////////////////
+    row_set_operation::row_set_operation(
+            std::vector<primitive_argument_type>&& operands,
+            std::string const& name, std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+    {}
+
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+
+        struct set_row : std::enable_shared_from_this<set_row>
+        {
+            set_row(std::string const& name, std::string const& codename)
+              : name_(name)
+              , codename_(codename)
+            {
+            }
+
+        protected:
+            std::string name_;
+            std::string codename_;
+
+        protected:
+            using arg_type = ir::node_data<double>;
+            using args_type = std::vector<arg_type>;
+
+            using storage0d_type = typename arg_type::storage0d_type;
+            using storage1d_type = typename arg_type::storage1d_type;
+            using storage2d_type = typename arg_type::storage2d_type;
+
+            std::vector<int> create_list_row_set(
+                int start, int stop, int step, int array_length) const
+            {
+                auto actual_start = 0;
+                auto actual_stop = 0;
+
+                if (start >= 0)
+                {
+                    actual_start = start;
+                }
+
+                if (start < 0)
+                {
+                    actual_start = array_length + start;
+                }
+
+                if (stop >= 0)
+                {
+                    actual_stop = stop;
+                }
+
+                if (stop < 0)
+                {
+                    actual_stop = array_length + stop;
+                }
+
+                std::vector<int> result;
+
+                if (step > 0)
+                {
+                    for (int i = actual_start; i < actual_stop; i += step)
```

Comment:
Would it be possible to `reserve()` the appropriate amount of entries for this vector? Something like `result.reserve((actual_stop-actual_start+step)/step);`?